src/GraphQL/Internal/Validation.hs

```diff
@@ -64,7 +64,7 @@ validate :: Alternative m => AST.Document -> m ValidDocument
 validate = pure . Valid
 
 data ValidationError
-  = DuplicateOperation AST.Name
+  = DuplicateOperation (Maybe AST.Name)
```

Comment:
An "operation" in a GraphQL document is a query or a mutation. e.g.

```
query foo { ... }
mutation bar { ... }
```

You cannot have more than one operation with the same name.

`DuplicateOperation` is the error that means that you have more than one operation with the same name.

Because GraphQL allows anonymous queries, you can theoretically write:

```
{
  me {
    name
  }
}
{
  me {
    age
  }
}
```

But this is also invalid, because it has more than one anonymous query.

This validation code takes the cheap approach of treating that as a `DuplicateOperation Nothing`. I could adjust the data type to have separate errors for these cases, if you'd like